keras_preprocessing/image/dataframe_iterator.py

```diff
@@ -173,6 +171,13 @@ def _check_params(self, df, x_col, y_col, weight_col, classes):
         if self.class_mode not in self.allowed_class_modes:
             raise ValueError('Invalid class_mode: {}; expected one of: {}'
                              .format(self.class_mode, self.allowed_class_modes))
+        # check that y_col has several column names if class_mode is multi_output
+        if (self.class_mode == 'multi_output') and not isinstance(y_col, list):
+            raise TypeError(
+                'If class_mode="{}", y_col must be a list. Received {}. '
+                'For regression use class_mode="raw".'
```

Comment:
I don't think this line is necessary or otherwise could be rephrased better, This statement may confuse people if they're trying to fit a multi-output regression model.

Reply:
I think the statements under `if class_mode not in ["input", "multi_output", "raw", None]:` at Line 147 can be moved under Line 135 which does the same check. Not necessarily to be fixed in this PR though.

Reply:
they are the same checks indeed but unfortunately they cannot be under the same if statement because of the split functionality. As I mentioned before in other PR's the problem here are the naming of variables which is confusing, in the first if statement we collect the classes and in the second we collect the labels (but both have the classes variables name).